pallene/ast.lua

```diff
@@ -63,7 +63,7 @@ declare_type("Exp", {
     Float      = {"loc", "value"},
     String     = {"loc", "value"},
     Initlist   = {"loc", "fields"},
-    Lambda     = {"loc", "arg_decls", "body"},
+    Lambda     = {"loc", "arg_decls", "type", "body"},
```

Comment:
Lambdas should not carry a type. Don't be destracted by that evil decl that we used to add to give to the functions. It's no longer present after the refactoring, for good reason :)

Reply:
I ended up removing that, but now a call to `FuncExp` returns a lambda + type.
```lua
 return ast.Exp.Lambda(name_exp.loc, params, block), func_typ
```
Since I'm not supposed to add new ast nodes in this PR, is it okay to not carry types in a lambda?
I wonder how the type checker will access the type tags for function expressions that are just an `ast.Exp.Lambda` nodes by themselves. (Since the only other node representing a function is in `ast.Stat`).

Reply:
A lambda by itself cannot be type checked, similarly to how a table literal by itself cannot be type checked. This is why check exp synthesise raises an error if it hits a lambda.

A lambda always has to appear in a context where the expected type type is known.  (Check exp verify)

Reply:
Aah, so the annotation is used. Gotcha!